src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java

```diff
@@ -62,8 +77,11 @@ protected Void doInBackground() throws Exception {
 			HttpMessage msg = this.httpMessage.cloneRequest();
 
 			try {
-				msg.getRequestHeader().setHeader(HttpHeader.COOKIE, null);
+				// SB: Only remove the cookie that you want to analyse
+				// msg.getRequestHeader().setHeader(HttpHeader.COOKIE, null);
+				msg.getRequestHeader().setCookieParams(cookies);
 				this.getHttpSender().sendAndReceive(msg, true);
+				Thread.sleep (10); // Not to flood all the requests

```

Comment:
Is this really necessary?

If there's a timing issue (un-related to the cookie changes) then this should be a separate issue/pull.